### ISSUE TITLE
Angular 10: Runtime Error Error: Uncaught (in promise): TypeError: Cannot read property 'firstCreatePass' of null

### ISSUE THREAD

**Issue by pspkshah** [angular/angular#37941]
After upgrading from Angular 9 to 10, I am getting below error on runtime for a child component.

I can reach component, but it fails to bind for [(ngModel)] property and giving below error

**Stack Trace**

Error: Uncaught (in promise): TypeError: Cannot read property 'firstCreatePass' of null
TypeError: Cannot read property 'firstCreatePass' of null
    at providersResolver (core.js:27050)
    at Object.definition.providersResolver (core.js:27382)
    at resolveDirectives (core.js:7869)
    at elementStartFirstCreatePass (core.js:13817)
    at Module.ɵɵelementStart (core.js:13855)
    at ProductFormComponent_Template (product-form.component.html:17)
    at executeTemplate (core.js:7329)
    at renderView (core.js:7138)
    at renderComponent (core.js:8382)
    at renderChildComponents (core.js:6997)
    at resolvePromise (zone-evergreen.js:798)
    at resolvePromise (zone-evergreen.js:750)
    at zone-evergreen.js:860
    at ZoneDelegate.invokeTask (zone-evergreen.js:399)
    at Object.onInvokeTask (core.js:27126)
    at ZoneDelegate.invokeTask (zone-evergreen.js:398)
    at Zone.runTask (zone-evergreen.js:167)
    at drainMicroTaskQueue (zone-evergreen.js:569)
    at invokeTask (zone-evergreen.js:484)
    at ZoneTask.invoke (zone-evergreen.js:469)

**Comment by JoostK** [comment]
Heya,

There might be a real issue here, however there's insufficient information provided for us to investigate the issue. Please provide a reproduction in StackBlitz or a GH repo.

**Comment by seeliang** [comment]
having the same issue when moving sub project to yarn workspaces on angular 9

**Comment by snortblt** [comment]
I ran into this when I converted an app containing custom form components to a library + app in angular 10

**Comment by chriscurnow** [comment]
Exactly the same here.

If I leave the library in the `dist/` folder it works fine.

When I use npm link to use the library in another app I get this error.

Angular v11

**Comment by JohnMunsch** [comment]
I had this problem in spades until I upgraded both the library I was working on and the project which was attempting to consume it to v11.2.2. After that it started working without the error seen above as well as another one not mentioned where it said that there were multiple components with the same selector (for example, "lib-micro-charts") for any component I tried to use from the library.

I knew that there was no duplication because I could alter the selector in the one component I was trying to use to any random thing and it still said there were duplicates. Again... until I upgraded. And then poof, suddenly I had no more problems.
